src/ejabberd_auth.erl

```diff
@@ -30,6 +30,16 @@
 -export([start/0,
          start/1,
          stop/1,
+         set_generic_opts/3,
+         get_generic_opt/2,
+         get_generic_opt/3,
+         get_jwt_username_key/1,
```

Comment:
Lines 36-42 are too specific, they shouldn't be provided as API here. 33-35 should be sufficient for `ejabberd_auth_*` modules. `ejabberd_auth` shouldn't be aware about specific possible config keys inside auth_opts.